planner/languages/java/java_planner.go

```diff
@@ -17,9 +33,64 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	return false
+	// Checking for pom.xml (maven) only for now
+	// TODO: add build.gradle file detection
+	pomXMLPath := filepath.Join(srcDir, "pom.xml")
+	return plansdk.FileExists(pomXMLPath)
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
-	return &plansdk.Plan{}
+	plan := &plansdk.Plan{
+		DevPackages: []string{
+			defaultMaven,
+		},
+	}
+	javaPkg, err := getJavaPackage(srcDir)
+	if err != nil {
+		return plan.WithError(err)
+	}
+	plan.DevPackages = append(plan.DevPackages, javaPkg)
+	return plan
+}
+
+func getJavaPackage(srcDir string) (string, error) {
+	pomXMLPath := filepath.Join(srcDir, "pom.xml")
+	javaVersion, err := parseJavaVersion(pomXMLPath)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	v, ok := jVersionMap[javaVersion]
+	if ok {
+		return v, nil
+	} else {
+		return defaultJava, nil
+	}
+}
+
+func parseJavaVersion(pomXMLPath string) (int, error) {
+	parsedVersion, err := parseXML(pomXMLPath)
+	if err != nil {
+		return 0, errors.WithMessage(err, "error parsing java version from pom file")
+	}
+	return parsedVersion, nil
+}
+
+func parseXML(pomXMLPath string) (int, error) {
```

Comment:
these two functions can now be collapsed, i think

Reply:
@savil added the `README.md`. I'll add the specific java version check in the PR for build.